src/net/ossl.cc

```diff
@@ -816,8 +854,30 @@ class session : public enable_shared_from_this<session>, public session_impl {
 
 
 private:
+    std::shared_ptr<X509> get_peer_certificate() const {
+        auto res = SSL_get_verify_result(_ssl.get());
+        auto peer_cert = make_shared_x509_cert(SSL_get1_peer_certificate(_ssl.get()));
+        if (!peer_cert && res != 0) {
+            // If there was a verification error, must obtain cert through cert_creds instance
+            peer_cert = _creds->get_last_cert();
+            if (!peer_cert) {
+                return nullptr;
+            }
```

Comment:
This appears to be a noop, you can just drop out of the scope or return directly.